Description: Disable automatic version check and selfupdate.
  Warn and ask on manual selfupdate.
Author: Jens Reyer <jre.winesim@gmail.com>
Forwarded: not-needed
Last-Update: 2019-12-29

--- a/src/winetricks
+++ b/src/winetricks
@@ -1095,6 +1095,18 @@ winetricks_check_update_availability()
         w_warn "You don't have the proper permissions to run this command. Try again with sudo or as root."
         exit
     fi
+
+    w_warn "This will install Winetricks directly from its original developers.  Debian has no control over that version."
+    printf %s "To continue press Y, to abort N, then Enter: "
+    read -r debresponse
+    if test "$debresponse" = Y || test "$debresponse" = y; then
+        true
+    elif test "$debresponse" = N || test "$debresponse" = n; then
+        exit
+    else
+        w_warn "Please press Y or N.  Aborting"
+        exit
+    fi
 }
 
 winetricks_selfupdate()
@@ -1136,6 +1148,8 @@ winetricks_selfupdate()
     w_try chmod +x "$0"
 
     w_warn "Update finished! The current version is $($0 -V). Use 'winetricks --update-rollback' to return to the previous version."
+    w_warn "Winetricks is no more controlled by Debian but by the Winetricks project now."
+    w_warn "The next update of the winetricks package may or may not revert this change."
 
     exit
 }
@@ -3231,10 +3245,10 @@ winetricks_latest_version_check()
     if [ ! "$WINETRICKS_VERSION" = "${latest_version}" ] && [ ! "$WINETRICKS_VERSION" = "${latest_version}-next" ]; then
         if [ -f "${WINETRICKS_CONFIG}/enable-auto-update" ] ; then
             w_info "You are running winetricks-${WINETRICKS_VERSION}."
-            w_info "New upstream release winetricks-${latest_version} is available."
-            w_info "auto-update enabled: running winetricks_selfupdate"
-            winetricks_selfupdate
-        else
+            w_info "The automatic selfupdate is disabled in the Debian packages."
+            #winetricks_selfupdate
+        fi
+        #else
             case $LANG in
                 pl*)
                     w_warn "Korzystasz z winetricks-${WINETRICKS_VERSION}, a najnowszą wersją winetricks-${latest_version}!"
@@ -3249,7 +3263,7 @@ winetricks_latest_version_check()
                     w_warn "You should update using your distribution's package manager, --self-update, or manually."
                     ;;
             esac
-        fi
+        #fi
     fi
 }
 
@@ -5251,7 +5265,8 @@ winetricks_wine_setup()
         return
     fi
 
-    winetricks_latest_version_check
+    # In Debian we don't connect automatically to a remote site for privacy reasons.
+    #winetricks_latest_version_check
 
     # Overridden for windows
     W_ISO_MOUNT_ROOT=/mnt/winetricks
@@ -5421,7 +5436,7 @@ Options:
 -f,  --force           Don't check whether packages were already installed
     --gui             Show gui diagnostics even when driven by commandline
     --isolate         Install each app or game in its own bottle (WINEPREFIX)
-    --self-update     Update this application to the last version
+    --self-update     Update this application to the latest version (non-Debian)
     --update-rollback Rollback the last self update
 -k, --keep_isos       Cache isos (allows later installation without disc)
     --no-clean        Don't delete temp directories (useful during debugging)
